lib/dpl/provider/releases.rb

```diff
@@ -100,22 +100,23 @@ def push_app
         end
 
         files.each do |file|
-          already_exists = false
+          existing_url = nil
           filename = Pathname.new(file).basename.to_s
           api.release(release_url).rels[:assets].get.data.each do |existing_file|
             if existing_file.name == filename
-              already_exists = true
+              existing_url = existing_file.url
             end
           end
-          if already_exists
-            log "#{filename} already exists, skipping."
-          else
+          if !existing_url || (existing_url && options[:overwrite])
+            api.delete_release_asset(existing_url) if existing_url
```

Comment:
Added in 4464031; it required a little more refactoring to achieve without duplication.